File: sdk/java-sdk/src/main/java/com/evernym/verity/sdk/exceptions/WalletException.java
package com.evernym.verity.sdk.exceptions;

/**
 * Exception class that is thrown when wallet operations fail
 */
public class WalletException extends VerityException {
    /**
     * Constructs a new WalletException with the given message.
     *
     * @param message the given message for the cause of the exception.
     *
     * @see VerityException#VerityException(String)
     */
    public WalletException(String message) {super(message);}

    /**
     * Constructs a new InvalidMessageTypeException with the given message and cause.
     *
     * @param message the given message for the cause of the exception.
     * @param cause the given cause (Throwable) for the exception.
     *
     * @see VerityException#VerityException(String, Throwable)
     */
    public WalletException(String message, Throwable cause) {super(message, cause);}
}

File: samples/sdk/dotnet-example-app-tmp/VeritySDK/Exceptions/WalletException.cs
using System;

namespace VeritySDK.Exceptions
{
    ///// <summary>
    ///// Exception class that is thrown when the wallet is un-expectantly closed
    ///// </summary>
    public class WalletException : VerityException
    {
        /// <summary>
        /// Constructs a new exception with the given message.
        /// </summary>
        /// <param name="message">the given message for the cause of the exception.</param>
        public WalletException(string message) : base(message) { }

        /// <summary>
        /// Constructs a new exception with the given message and cause.
        /// </summary>
        /// <param name="message">the given message for the cause of the exception.</param>
        /// <param name="cause">the given cause for the exception.</param>
        public WalletException(String message, System.Exception cause) : base(message, cause) { }
    }
}
